flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManagerTest.java

```diff
@@ -99,56 +103,340 @@ void testGenerateInsertStatementWithDynamicOptions() {
+                        .tableOptions("partition.fields.day.date-formatter", "yyyy-MM-dd")
+                        .tableOptions("partition.fields.hour.date-formatter", "HH")
+                        .expectedRefreshPartition("day", "2023-12-31")
+                        .expectedRefreshPartition("hour", "23"),
+                TestSpec.create()
+                        .schedulerTime("2024-01-01 00:00:00")
+                        .freshness(IntervalFreshness.ofMinute("2"))
+                        .tableOptions("partition.fields.day.date-formatter", "yyyy-MM-dd")
+                        .tableOptions("partition.fields.hour.date-formatter", "HH")
+                        .expectedRefreshPartition("day", "2023-12-31")
+                        .expectedRefreshPartition("hour", "23"),
+                TestSpec.create()
+                        .schedulerTime("2024-01-01 00:00:00")
+                        .freshness(IntervalFreshness.ofMinute("4"))
+                        .tableOptions("partition.fields.day.date-formatter", "yyyy-MM-dd")
+                        .tableOptions("partition.fields.hour.date-formatter", "HH")
+                        .expectedRefreshPartition("day", "2023-12-31")
+                        .expectedRefreshPartition("hour", "23"),
+                TestSpec.create()
+                        .schedulerTime("2024-01-01 00:00:00")
+                        .freshness(IntervalFreshness.ofMinute("15"))
+                        .tableOptions("partition.fields.day.date-formatter", "yyyy-MM-dd")
+                        .tableOptions("partition.fields.hour.date-formatter", "HH")
+                        .expectedRefreshPartition("day", "2023-12-31")
+                        .expectedRefreshPartition("hour", "23"),
+                TestSpec.create()
+                        .schedulerTime("2024-01-01 00:00:00")
+                        .freshness(IntervalFreshness.ofMinute("1"))
+                        .tableOptions("partition.fields.day.date-formatter", "yyyy-MM-dd")
+                        .expectedRefreshPartition("day", "2023-12-31"),
+                TestSpec.create()
+                        .schedulerTime("2024-01-01 00:01:00")
+                        .freshness(IntervalFreshness.ofMinute("1"))
+                        .tableOptions("partition.fields.day.date-formatter", "yyyy-MM-dd")
+                        .expectedRefreshPartition("day", "2024-01-01"),
+
+                // Invalid test case.
+                TestSpec.create()
+                        .schedulerTime(null)
+                        .freshness(IntervalFreshness.ofDay("1"))
+                        .tableOptions("partition.fields.day.date-formatter", "yyyy-MM-dd")
+                        .tableOptions("partition.fields.hour.date-formatter", "HH")
+                        .errorMessage(
+                                "Scheduler time not properly set for periodic refresh of materialized table `catalog`.`database`.`table`."),
```

Comment:
I think this error message is not clear for use, we should emphasize the scheduler time is null.